test/e2e/apps/statefulset.go

```diff
@@ -1154,6 +1154,50 @@ var _ = SIGDescribe("StatefulSet", func() {
 		framework.ExpectNoError(err)
 		e2estatefulset.WaitForStatusAvailableReplicas(c, ss, 1)
 	})
+
+	ginkgo.It("AvailableReplicas should get updated accordingly when MinReadySeconds is enabled [Feature:StatefulSetMinReadySeconds] [alpha]", func() {
+		ssName := "test-ss"
+		headlessSvcName := "test"
+		// Define StatefulSet Labels
+		ssPodLabels := map[string]string{
+			"name": "sample-pod",
+			"pod":  WebserverImageName,
+		}
+		ss := e2estatefulset.NewStatefulSet(ssName, ns, headlessSvcName, 2, nil, nil, ssPodLabels)
+		ss.Spec.MinReadySeconds = 30
+		setHTTPProbe(ss)
+		ss, err := c.AppsV1().StatefulSets(ns).Create(context.TODO(), ss, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		e2estatefulset.WaitForStatusAvailableReplicas(c, ss, 0)
+		// let's check that the availableReplicas have still not updated
+		time.Sleep(5 * time.Second)
```

Comment:
I'd prefer not to use `time.Sleep` but rather add those extra 5s in the next `WaitFor...`, if needed.

Reply:
Although looking at `WaitForStatusAvailableReplicas` it waits up to 10mins, so you can just drop that bit.

Reply:
as mentioned in https://github.com/kubernetes/kubernetes/pull/104078#discussion_r709091077 we might observe 0 AvailableReplicas immediately and this ensures that we wait at least 5 seconds to check it still stays 0.

The next WaitForStatusAvailableReplicas check for Availability. This one checks for UnAvailability

Reply:
Oh, I see what you mean, I missed the initial `MinReadySeconds` being set, I assumed it's not.